server/src/main/java/org/elasticsearch/search/aggregations/metrics/max/MaxAggregator.java

```diff
@@ -43,16 +46,18 @@
 
     final ValuesSource.Numeric valuesSource;
     final DocValueFormat formatter;
+    final CheckedFunction<LeafReader, Number, IOException>  maxFunc;
 
     DoubleArray maxes;
 
     public MaxAggregator(String name, ValuesSource.Numeric valuesSource, DocValueFormat formatter,
             SearchContext context,
             Aggregator parent, List<PipelineAggregator> pipelineAggregators,
-            Map<String, Object> metaData) throws IOException {
+            Map<String, Object> metaData, CheckedFunction<LeafReader, Number, IOException> maxFunc) throws IOException {
```

Comment:
I think we should call this something other than `maxFunc` as it confuses what its actually doing (getting the max for the segment) from the normal running of the agg. Maybe calling it `segmentMaxFunction` would be clearer?

Reply:
++